src/main/java/org/alfresco/repo/jscript/ScriptNode.java

```diff
@@ -3113,10 +3120,18 @@ public ScriptThumbnail createThumbnail(String thumbnailName, boolean async, bool
             }
             else
             {
-                Action action = ThumbnailHelper.createCreateThumbnailAction(details, services);
-                
-                // Queue async creation of thumbnail
-                this.services.getActionService().executeAction(action, this.nodeRef, true, true);
+                RenditionDefinition2 renditionDefinition = renditionDefinitionRegistry2.getRenditionDefinition(thumbnailName);
+                if (renditionDefinition != null)

```

Comment:
Is it worth checking if renditionService2 is null as well? In case if it was not injected it a custom code.

Reply:
I think that is going too far.